confgenerator/config.go

```diff
@@ -346,7 +346,9 @@ type Logging struct {
 
 type LoggingReceiver interface {
 	Component
-	Components(tag string) []fluentbit.Component
+	// Components returns fluentbit components that implement this receiver.
+	// tag is the log tag that should be produced by those components, and platform is the target platform.
+	Components(tag string, platform string) []fluentbit.Component
```

Comment:
Adding a string parameter like this is a major change, and it's going to be even more painful if we have to add more later. If we decided we need to go down this road, we should add a `context.Context` parameter instead so that we only have to do this once.